src/EventHandler/BotApp.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+/**
+ * This file is part of MadelineProto.
+ * MadelineProto is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ * MadelineProto is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with MadelineProto.
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author    Amir Hossein Jafari <amirhosseinjafari8228@gmail.com>
+ * @copyright 2016-2023 Amir Hossein Jafari <amirhosseinjafari8228@gmail.com>
+ * @license   https://opensource.org/licenses/AGPL-3.0 AGPLv3
+ * @link https://docs.madelineproto.xyz MadelineProto documentation
+ */
+
+namespace danog\MadelineProto\EventHandler;
+
+use danog\MadelineProto\EventHandler\Media\Document;
+use danog\MadelineProto\EventHandler\Media\Photo;
+use danog\MadelineProto\MTProto;
+use JsonSerializable;
+use ReflectionClass;
+use ReflectionProperty;
+
+/**
+ * Represents information about a named bot web app.
+ */
+final class BotApp implements JsonSerializable
```

Comment:
What do you mean

Reply:
Since we have an abstract BotApp method, it's a good idea to also have a MadelineProto method that can fetch a bot app by its id, returning a BotApp object

Reply:
Well should I declare it in BotApp object or DialogBotAllowed?
and if I declare it in BotApp I can not have access to `$this->getClient`

Reply:
In MTProto.php

Reply:
getBotApp returns more information.

Should I create a new class or add properties in this class?

Reply:
Add more fields I think